crates/db_views/src/community/community_view.rs

```diff
@@ -151,40 +178,44 @@ impl CommunityQuery<'_> {
     // also hidden (based on profile setting)
     query = query.filter(instance_actions::blocked.is_null());
     query = query.filter(community_actions::blocked.is_null());
-    if !(o.local_user.show_nsfw(site) || o.show_nsfw) {
+    if !(o.local_user.show_nsfw(site) || o.show_nsfw.unwrap_or_default()) {
       query = query.filter(community::nsfw.eq(false));
     }
 
     query = o.local_user.visible_communities_only(query);
 
-    match o.sort.unwrap_or_default() {
-      Hot => query = query.order_by(community::hot_rank.desc()),
-      Comments => query = query.order_by(community::comments.desc()),
-      Posts => query = query.order_by(community::posts.desc()),
-      New => query = query.order_by(community::published.desc()),
-      Old => query = query.order_by(community::published.asc()),
-      Subscribers => query = query.order_by(community::subscribers.desc()),
-      SubscribersLocal => query = query.order_by(community::subscribers_local.desc()),
-      ActiveSixMonths => query = query.order_by(community::users_active_half_year.desc()),
-      ActiveMonthly => query = query.order_by(community::users_active_month.desc()),
-      ActiveWeekly => query = query.order_by(community::users_active_week.desc()),
-      ActiveDaily => query = query.order_by(community::users_active_day.desc()),
-      NameAsc => query = query.order_by(lower(community::name).asc()),
-      NameDesc => query = query.order_by(lower(community::name).desc()),
-    };
     // Filter by the time range
     if let Some(time_range_seconds) = o.time_range_seconds {
       query =
         query.filter(community::published.gt(now() - seconds_to_pg_interval(time_range_seconds)));
     }
 
-    let (limit, offset) = limit_and_offset(o.page, o.limit)?;
+    // Only sort by ascending for Old or NameAsc sorts.
+    let sort = o.sort.unwrap_or_default();
+    let sort_direction = asc_if(sort == Old || sort == NameAsc);
+
+    let mut pq = paginate(query, sort_direction, o.cursor_data, None, o.page_back);
+
+    pq = match sort {
+      Hot => pq.then_order_by(key::hot_rank),
+      Comments => pq.then_order_by(key::comments),
+      Posts => pq.then_order_by(key::posts),
+      New => pq.then_order_by(key::published),
+      Old => pq.then_order_by(key::published),
+      Subscribers => pq.then_order_by(key::subscribers),
+      SubscribersLocal => pq.then_order_by(key::subscribers_local),
+      ActiveSixMonths => pq.then_order_by(key::users_active_half_year),
+      ActiveMonthly => pq.then_order_by(key::users_active_month),
+      ActiveWeekly => pq.then_order_by(key::users_active_week),
+      ActiveDaily => pq.then_order_by(key::users_active_day),
+      // TODO the lower function doesn't work for these
```

Comment:
Create a `LowerKey` struct, exactly the same as `ReverseTimestampKey` other than what SQL function it calls.

Reply:
Nice, thx.